src/core/lib/security/credentials/tls/grpc_tls_crl_provider.cc

```diff
@@ -159,11 +159,7 @@ absl::StatusOr<std::shared_ptr<CrlProvider>> CreateDirectoryReloaderCrlProvider(
       MakeDirectoryReader(directory));
   // This could be slow to do at startup, but we want to
   // make sure it's done before the provider is used.
-  absl::Status initial_status = provider->Update();
-  if (!initial_status.ok()) {
-    return initial_status;
-  }
-  provider->ScheduleReload();
+  provider->OnNextUpdateTimer();
```

Comment:
Suggest moving this into the `DirectoryReloaderCrlProvider` ctor, so that the code here can just say `return std:::make_shared<DirectoryReloaderCrlProvider>(...)`.  That way, `OnNextUpdateTimer()` can stay private within `DirectoryReloaderCrlProvider`.

Reply:
Ah I like this much better

Reply:
After trying to implement I don't think we can do this - we can't use `shared_from_this` in a call chain from the ctor

Reply:
Oh, interesting.

Okay, in that case, please at least rename `OnNextUpdateTimer()` to `UpdateAndStartTimer()`, to reflect the fact that it's now being called in a place that is not a timer firing.

Reply:
Done